analyzers/src/SonarAnalyzer.CSharp/Extensions/TypeDeclarationSyntaxExtensions.cs

```diff
@@ -34,5 +34,30 @@ public static class TypeDeclarationSyntaxExtensions
             methodDeclaration.DescendantNodes()
                              .Where(member => member.IsKind(SyntaxKindEx.LocalFunctionStatement))
                              .Select(member => MethodDeclarationFactory.Create(member));
+
+        public static IMethodSymbol PrimaryConstructor(this TypeDeclarationSyntax typeDeclaration, SemanticModel semanticModel)
+        {
+            if (ParameterList(typeDeclaration) is { } parameterList)
+            {
+                return parameterList is { Parameters: { Count: > 0 } parameters }
+                    ? semanticModel.GetDeclaredSymbol(parameters[0])?.ContainingSymbol as IMethodSymbol
+                    : semanticModel.GetDeclaredSymbol(typeDeclaration).GetMembers(".ctor").OfType<IMethodSymbol>().FirstOrDefault(m => m is
+                    {
+                        MethodKind: MethodKind.Constructor,
+                        Parameters.Length: 0,
```

Comment:
Educational: why do we want a ctor with 0 parameters?

Reply:
We are looking for a primary constructor, whose parameter count is zero according to line 42.